### PR TITLE
Modified version numbers with variables at 3.12

### LINKED ISSUE #2301: Specify versions in older documentation
If a client is looking at an older version of our documentation on purpose. Wanting to install an older version. Our documentation will give them commands like this: 

```yum install wazuh-manager```

Which will install the latest version. 

So for example, in the documentation for 3.10, the command should be: 

```yum install wazuh-manager-3.10.2-1```

This will point to the latest release of the 3.10 version. But this is fine because seeing this, if the client wanted the 3.10.0 release, they can easily change it knowing the command. It is intuitive. 

It can also have indirect advantages, like clients that have not seen the top banner advising that they are on an outdated section of the documentation, they could see that the version on the command is not the one they are looking for. 

This affects many places in the documentation. It is not an urgent matter, but it could be interesting to have it implemented in the future.

### PR DESCRIPTION
**Pull request by jchermoso**
Hello team!

This closes #2301 of 3.12 

## Description
Modified version references with version variables

## Checks
- [x] It compiles without warnings.
- [x] Spelling and grammar. 
- [x] Used impersonal speech. 
- [x] Used uppercase only on nouns. 
- [x] Updated the `redirect.js` script if necessary (check [this guide](https://github.com/wazuh/wazuh-documentation/blob/master/NEW_RELEASE.md)).

Best regards,

Juan Carlos

### CODE REVIEW

File: source/installation-guide/installing-wazuh-agent/deployment_variables/linux/deployment_variables_yum.rst [hunk 1]
```diff
@@ -47,41 +47,41 @@ Examples:
 .. code-block:: console
 
      # WAZUH_MANAGER="10.0.0.2" WAZUH_REGISTRATION_PASSWORD="TopSecret" \
-          WAZUH_AGENT_NAME="yum-agent" yum install wazuh-agent
+          WAZUH_AGENT_NAME="yum-agent" yum install wazuh-agentC
```
Comment (alberpilot) [inline source/installation-guide/installing-wazuh-agent/deployment_variables/linux/deployment_variables_yum.rst:50]: I don't think that we have a `wazuh-agentC`. Please be careful

File: source/upgrade-guide/upgrading/different_major.rst [hunk 2]
```diff
@@ -46,7 +46,7 @@ Upgrade Wazuh manager
     .. code-block:: console
 
       # yum install wazuh-manager
-
+
```
Comment (alberpilot) [inline source/upgrade-guide/upgrading/different_major.rst:49]: What about this tab?

**Review by alberpilot — CHANGES_REQUESTED** [review]
Please review the requested changes

File: source/installation-guide/installing-wazuh-agent/deployment_variables/linux/deployment_variables_yum.rst (hunk 1 above)
Comment (jchermoso) [reply source/installation-guide/installing-wazuh-agent/deployment_variables/linux/deployment_variables_yum.rst:50]: My bad, sorry.

**Review by alberpilot — APPROVED** [review]

**Pull request merged**